servers/zms/src/test/java/com/yahoo/athenz/zms/store/jdbc/JDBCConnectionTest.java

```diff
@@ -9358,4 +9358,37 @@ public void testListRolesWithUserAuthorityRestrictionsException() throws Excepti
         }
         jdbcConn.close();
     }
+
+    @Test
+    public void testGetDomainFromTemplateName() throws Exception {
+        Map<String, Integer> templateDetails = new HashMap<>();
+        templateDetails.put("aws", 1);
+        templateDetails.put("aws_bastion", 2);
+        String domainName = "testdom";
+        String templateName1 = "testtemplate";
+        String templateName2 = "testtemplate2";
+
+        JDBCConnection jdbcConn = new JDBCConnection(mockConn, true);
+
+        Mockito.when(mockResultSet.getString("name"))
+                .thenReturn(domainName); // domain name
+        Mockito.when(mockResultSet.getString("template"))
+                .thenReturn(templateName1); // template name
+        Mockito.when(mockResultSet.getString("name"))
+                .thenReturn(domainName);
+        Mockito.when(mockResultSet.getString("template"))
+                .thenReturn(templateName2);
+
+        Mockito.when(mockResultSet.next())
+                .thenReturn(true).thenReturn(true).thenReturn(false); // this one is for domain id
```

Comment:
not sure what the comment at the end about domain id is referring to

Reply:
sorry my bad. removed it